tools/topology/m4/pcm.m4

```diff
@@ -74,6 +74,20 @@ define(`PCM_CAPABILITIES',
 dnl PCM_PLAYBACK_ADD(name, pcm_id, playback)
 define(`PCM_PLAYBACK_ADD',
 `ifelse(`$#', `3',
+PCM_PLAYBACK_ADD_COMMON($1, $2, $3, false),
+`fatal_error(`Invalid parameters ($#) to PCM_PLAYBACK_ADD')')'
+)
+
+dnl PCM_PLAYBACK_COMPR_ADD(name, pcm_id, playback)
+define(`PCM_PLAYBACK_COMPR_ADD',
+`ifelse(`$#', `3',
+PCM_PLAYBACK_ADD_COMMON($1, $2, $3, true),
+`fatal_error(`Invalid parameters ($#) to PCM_PLAYBACK_COMPR_ADD')')'
```

Comment:
Could be more specific in the message: Invalid parameter count.

Reply:
Ah, noticed the preexisting error messages aren't this specific. A pass-by refactor might be nice but I won't mandate it.

Reply:
Yes, I kept the original error message. This could be refactor later. Marking the comment as resolved.